### PR TITLE
Utility function - immediate children on any Node (or) ASTNode

### PR DESCRIPTION
**Pull request by TilakMaddy**
.

### CODE REVIEW

**Comment by TilakMaddy** [comment]
Again I'll merge the lint corrections from cross compile ci cd draft : )

File: aderyn_core/src/detect/experimental/immediate_children.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            .filter(|x| x.name.contains("increment"))
+            .take(1)
+            .next()
+            .unwrap();
+
+        if let Some(children) = inc.body.as_ref().unwrap().immediate_children(context) {
+            for child in children {
+                assert!(
+                    child.node_type() == NodeType::IfStatement,
+                    "Only if statement should be caught in function body's immediate children!"
```
Comment (TilakMaddy) [inline aderyn_core/src/detect/experimental/immediate_children.rs:44]: Here, `inc.body` is a function block:
That has a for loop and an if statement nested in it. However, when we call `immediate_children(context)` on the `&Block` node, we only get the for loop

File: aderyn_core/src/detect/experimental/immediate_children.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        let mut detector = ImmediateChildrenDemonstrator::default();
+        let found = detector.detect(&context).unwrap();
+        assert!(found);
+
+        println!("{:?}", detector.instances());
+        println!(
+            "Total number of instances: {:?}",
+            detector.instances().len()
+        );
+        assert!(detector.instances().len() == 1);
```
Comment (TilakMaddy) [inline aderyn_core/src/detect/experimental/immediate_children.rs:96]: Verified here. Only 1 ast node is caught and that's the for loop

**Comment by TilakMaddy** [comment]
Contract we're talking about here is `ParentChainContract.sol`

File: aderyn_core/src/detect/experimental/immediate_children.rs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            .function_definitions()
+            .into_iter()
+            .filter(|x| x.name.contains("increment"))
+            .take(1)
+            .next()
+            .unwrap();
+
+        if let Some(children) = inc.body.as_ref().unwrap().children(context) {
+            for child in children {
+                assert!(
```
Comment (TilakMaddy) [inline aderyn_core/src/detect/experimental/immediate_children.rs:42]: I changed the method name to just `children()` I think it's more legible and intuitive (like you would start typing ".children" after the node)

**Review by alexroan — APPROVED** [review]

**Pull request merged**